### ISSUE TITLE
bugprone-unchecked-optional-access false positive

### ISSUE THREAD

**Issue by alexolog** [llvm/llvm-project#86185]
Code:
```
    if (auto error = checkMsg(msg, fieldMask, {}); error.has_value())
        throw std::runtime_error(*error);
```

Diagnostic:
```
protobuf_validation.h:69:35: warning: unchecked access to optional value [bugprone-unchecked-optional-access]
   69 |         throw std::runtime_error(*error);
      |                                   ^
```

LLVM 17.0.6 on Mac.

**Comment by EugeneZelenko** [comment]
Could you please try 18 or `main` branch?

**Comment by alexolog** [comment]
> Could you please try 18 or `main` branch?

As soon as it gets on homebrew :)

**Comment by chrchr-github** [comment]
Does not immediately repro with trunk:
https://godbolt.org/z/qobYKv1zn

**Comment by chrchr-github** [comment]
FWIW I'm not sure that these issues are all the same...
